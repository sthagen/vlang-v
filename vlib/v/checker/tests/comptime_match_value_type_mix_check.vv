module main

fn type_check_in_a_value_match() {
	x := 100
	$match x {
		100 {
			println('value check')
		}
		int {
			println('type check')
		}
	}
}

fn value_check_in_a_type_match() {
	x := 100
	$match x {
		int {
			println('type check')
		}
		100 {
			println('value check')
		}
	}
}
