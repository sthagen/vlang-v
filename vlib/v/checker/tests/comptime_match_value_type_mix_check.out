vlib/v/checker/tests/comptime_match_value_type_mix_check.vv:9:3: error: can not matching a type in a value `$match`
    7 |             println('value check')
    8 |         }
    9 |         int {
      |         ~~~
   10 |             println('type check')
   11 |         }
vlib/v/checker/tests/comptime_match_value_type_mix_check.vv:21:3: error: can not matching a value in a type `$match`
   19 |             println('type check')
   20 |         }
   21 |         100 {
      |         ~~~
   22 |             println('value check')
   23 |         }
